apps/gp2-server/test/routes/working-group.route.test.ts

```diff
@@ -120,4 +118,138 @@ describe('/working-groups/ route', () => {
       );
     });
   });
+  describe('PUT /working-group/{working_group_id}/resources', () => {
+    test('Should return the results correctly', async () => {
+      const loggedInUserId = '11';
+      const loggedUser: gp2.UserResponse = {
+        ...getUserResponse(),
+        id: loggedInUserId,
+        role: 'Administrator',
+      };
+      const getLoggedUser = jest.fn().mockReturnValue(loggedUser);
+      const authHandlerMock: AuthHandler = (req, _res, next) => {
+        req.loggedInUser = getLoggedUser();
+        next();
+      };
+      const appWithUser = appFactory({
+        workingGroupController: workingGroupControllerMock,
+        authHandler: authHandlerMock,
+        logger: loggerMock,
+      });
+      workingGroupControllerMock.update.mockResolvedValueOnce(
+        getWorkingGroupResponse(),
+      );
+      workingGroupControllerMock.fetchById.mockResolvedValueOnce(
+        getWorkingGroupResponse(),
+      );
+
+      const resources: gp2.Resource[] = [{ title: 'a resource', type: 'Note' }];
+      const response = await supertest(appWithUser)
+        .put('/working-group/23/resources')
+        .send(resources);
+
+      expect(response.status).toBe(200);
+      expect(response.body).toEqual(getWorkingGroupResponse());
+      expect(workingGroupControllerMock.update).toBeCalledWith(
+        '23',
+        { resources },
+        '11',
+      );
+    });
+
+    test('Should return 404 when working group doesnt exist', async () => {
+      const loggedInUserId = '11';
+      const loggedUser: gp2.UserResponse = {
+        ...getUserResponse(),
+        id: loggedInUserId,
+        role: 'Administrator',
+      };
+      const getLoggedUser = jest.fn().mockReturnValue(loggedUser);
+      const authHandlerMock: AuthHandler = (req, _res, next) => {
+        req.loggedInUser = getLoggedUser();
+        next();
+      };
+      const appWithUser = appFactory({
+        workingGroupController: workingGroupControllerMock,
+        authHandler: authHandlerMock,
+        logger: loggerMock,
+      });
+      workingGroupControllerMock.fetchById.mockResolvedValueOnce(
+        getWorkingGroupResponse(),
+      );
+      workingGroupControllerMock.update.mockRejectedValueOnce(Boom.notFound());
+
+      const response = await supertest(appWithUser)
+        .put('/working-group/123/resources')
+        .send([]);
+
+      expect(response.status).toBe(404);
+    });
+
+    test.each(gp2.userRoles.filter((role) => role !== 'Administrator'))(
+      'Should return 403 if the user is not in the Administrator role - %s',
+      async (role) => {
+        const loggedInUserId = '11';
+        const loggedUser: gp2.UserResponse = {
+          ...getUserResponse(),
+          id: loggedInUserId,
+          role,
+        };
+        const getLoggedUser = jest.fn().mockReturnValue(loggedUser);
```

Comment:
looks like a lot of repetition, can you move out the common stuff into describe or beforeEach/beforeAll?

Reply:
done